File: samples/JavaSamples/src/main/java/com/mku/salmon/samples/samples/DataSample.java
package com.mku.salmon.samples.samples;

import com.mku.convert.BitConverter;
import com.mku.salmon.Decryptor;
import com.mku.salmon.Encryptor;
import com.mku.salmon.Generator;
import com.mku.salmon.streams.EncryptionFormat;

import java.io.IOException;
import java.util.Arrays;

public class DataSample {

    public static byte[] encryptData(byte[] data, byte[] key, byte[] integrityKey, int threads) throws IOException {
        System.out.println("Encrypting bytes: " + BitConverter.toHex(Arrays.copyOf(data, 24)) + "...");

        // Always request a new random secure nonce.
        byte[] nonce = Generator.getSecureRandomBytes(8);

        Encryptor encryptor = new Encryptor(threads);
        byte[] encData = encryptor.encrypt(data, key, nonce, EncryptionFormat.Salmon,
                integrityKey != null, integrityKey);
        encryptor.close();

        System.out.println("Bytes encrypted: " + BitConverter.toHex(Arrays.copyOf(encData, 24)) + "...");
        return encData;
    }

    public static byte[] decryptData(byte[] data, byte[] key, byte[] integrityKey, int threads) throws IOException {
        System.out.println("Decrypting bytes: " + BitConverter.toHex(Arrays.copyOf(data, 24)) + "...");

        Decryptor decryptor = new Decryptor(threads);
        byte[] decBytes = decryptor.decrypt(data, key, null, EncryptionFormat.Salmon,
                integrityKey != null, integrityKey);
        decryptor.close();

        System.out.println("Bytes decrypted: " + BitConverter.toHex(Arrays.copyOf(decBytes, 24)) + "...");
        return decBytes;
    }
}

File: samples/CSharpSamples/CSharpSamples/Samples/DataSample.cs
using Mku.Salmon.Streams;
using BitConverter = Mku.Convert.BitConverter;

namespace Mku.Salmon.Samples.Samples;

class DataSample
{

    public static byte[] EncryptData(byte[] data, byte[] key, byte[] integrityKey, int threads)
    {
        Console.WriteLine("Encrypting bytes: " + BitConverter.ToHex(data.Take(24).ToArray()) + "...");

        // Always request a new random secure nonce.
        byte[] nonce = Generator.GetSecureRandomBytes(8);

        Encryptor encryptor = new Encryptor(threads);
        byte[] encData = encryptor.Encrypt(data, key, nonce, EncryptionFormat.Salmon,
                                             integrityKey != null, integrityKey);
        encryptor.Close();

        Console.WriteLine("Bytes encrypted: " + BitConverter.ToHex(encData.Take(24).ToArray()) + "...");
        return encData;
    }

    public static byte[] DecryptData(byte[] data, byte[] key, byte[] integrityKey,int threads)
    {
        Console.WriteLine("Decrypting bytes: " + BitConverter.ToHex(data.Take(24).ToArray()) + "...");

        Decryptor decryptor = new Decryptor(threads);
        byte[] decBytes = decryptor.Decrypt(data, key, null, EncryptionFormat.Salmon,
                                             integrityKey != null, integrityKey);
        decryptor.Close();

        Console.WriteLine("Bytes decrypted: " + BitConverter.ToHex(decBytes.Take(24).ToArray()) + "...");
        return decBytes;
    }
}